lib/experimental/class-theme-json-webfonts-handler.php

```diff
@@ -0,0 +1,537 @@
+					// Add data URIs first.
+					if ( 0 === strpos( trim( $url ), 'data:' ) ) {
+						$src_ordered[] = array(
+							'url'    => $url,
+							'format' => 'data',
+						);
+						continue;
+					}
+					$format         = pathinfo( $url, PATHINFO_EXTENSION );
+					$src[ $format ] = $url;
+				}
+
+				// Add woff2.
+				if ( ! empty( $src['woff2'] ) ) {
+					$src_ordered[] = array(
+						'url'    => $src['woff2'],
+						'format' => 'woff2',
+					);
+				}
+
+				// Add woff.
+				if ( ! empty( $src['woff'] ) ) {
+					$src_ordered[] = array(
+						'url'    => $src['woff'],
+						'format' => 'woff',
+					);
+				}
+
+				// Add ttf.
+				if ( ! empty( $src['ttf'] ) ) {
+					$src_ordered[] = array(
+						'url'    => $src['ttf'],
+						'format' => 'truetype',
+					);
+				}
+
+				// Add eot.
+				if ( ! empty( $src['eot'] ) ) {
+					$src_ordered[] = array(
+						'url'    => $src['eot'],
+						'format' => 'embedded-opentype',
+					);
+				}
+
+				// Add otf.
+				if ( ! empty( $src['otf'] ) ) {
+					$src_ordered[] = array(
+						'url'    => $src['otf'],
+						'format' => 'opentype',
+					);
+				}
+				$webfont['src'] = $src_ordered;
+
+				return $webfont;
+			}
+
+			/**
+			 * Builds the font-family's CSS.
+			 *
+			 * @since 6.0.0
+			 *
+			 * @param array $webfont Webfont to process.
+			 *
+			 * @return string This font-family's CSS.
+			 */
+			private function build_font_face_css( array $webfont ) {
+				$css = '';
+
+				// Wrap font-family in quotes if it contains spaces.
+				if (
+					str_contains( $webfont['font-family'], ' ' ) &&
+					! str_contains( $webfont['font-family'], '"' ) &&
+					! str_contains( $webfont['font-family'], "'" )
+				) {
+					$webfont['font-family'] = '"' . $webfont['font-family'] . '"';
+				}
+
+				foreach ( $webfont as $key => $value ) {
+					/*
+					 * Skip "provider", since it's for internal API use,
+					 * and not a valid CSS property.
+					 */
+					if ( 'provider' === $key ) {
+						continue;
+					}
+
+					// Compile the "src" parameter.
+					if ( 'src' === $key ) {
+						$value = $this->compile_src( $webfont['font-family'], $value );
+					}
+
+					// If font-variation-settings is an array, convert it to a string.
+					if ( 'font-variation-settings' === $key && is_array( $value ) ) {
+						$value = $this->compile_variations( $value );
+					}
+
+					if ( ! empty( $value ) ) {
+						$css .= "$key:$value;";
+					}
+				}
+
+				return $css;
+			}
+
+			/**
+			 * Compiles the `src` into valid CSS.
+			 *
+			 * @since 6.0.0
+			 *
+			 * @param string $font_family Font family.
+			 * @param array  $value       Value to process.
+			 *
+			 * @return string The CSS.
+			 */
+			private function compile_src( $font_family, array $value ) {
+				$src = "local($font_family)";
+
+				foreach ( $value as $item ) {
+
+					if ( 0 === strpos( $item['url'], get_site_url() ) ) {
+						$item['url'] = wp_make_link_relative( $item['url'] );
+					}
```

Comment:
```suggestion
					if (
						str_starts_with( $item['url'], site_url() )
						|| str_starts_with( $item['url'], home_url() )
					) {
						$item['url'] = wp_make_link_relative( $item['url'] );
					}
```

It's possible themes can live outside of the `site_url` if the config is something like:
* Home URL: example.com
* Site URL: example.com/wp
* Content URL: example.com/content

I'm suggesting checking for both as the site and home urls can technically be on different domains. 

The shorter function names also return so that's eight bytes that can be saved by dropping the gets

Reply:
Done https://github.com/WordPress/gutenberg/pull/40493/commits/66926e28c600e9eabb3701ec85e9af27bde0b129